src/ui/states/state-needs-payment-info.svelte

```diff
@@ -67,20 +67,32 @@
   });
 
   const handleContinue = async () => {
-    if (processing || !stripe || !safeElements) return;
+    if (processing || !stripe || !safeElements || !clientSecret) return;
 
     processing = true;
 
+    const isSetupIntent = clientSecret.startsWith("seti_");
```

Comment:
I'm not sure I like this implicit meaning... might be better to have a dedicated parameter to know what to use? Probably not a blocker since this is internal though.

Reply:
I'll create a ticket to make it more explicit.